articles/js-execution-under-the-hood/index.md

```diff
@@ -0,0 +1,167 @@
+    alt: JS execution under the hood
+---
+
+### Introduction
+JavaScript is a single-threaded language, where only one command executes at a time. And, it has a Synchronous model of execution, where each line executes line by line, top to bottom.
+
+In this guide, you get to understand what happens as the JavaScript engine threads down your code.
+
+## Prerequisites
+
+Before you begin this guide you'll need the following:
+
+- A basic understanding of JavaScript Functions
+
+## Step 1 — What Happens When JavaScript Executes My Code?
+
+```javascript 
+1. const num = 3;
+2. function multiplyBy2 (inputNumber){
+3.     const result = inputNumber*2;
+4.     return result;
+5.     }
+6. const name = "Will"
+```
+
+In this step, you will walk through the above lines of code like the JavaScript engine would.
+
+Why?
+
+To understand what happens when you execute the code above.
+
+First, as soon as you start executing your code, there are two halves to that process:
+
+1. One, the process of stepping through your code line by line, top to bottom. Referred to as the **Thread Of Execution**.
+2. Two, the storage of the bits of data that you announce as you go through your code's execution in live **Memory**. Referred to as the **Variable Environment**.
+
+Together, known as an **Execution Context**. Where a context is a space to do something. In this case, it is a space where you execute code, hence an execution context.
+
+There are two types of Execution contexts:
+
+1. Global Execution Context.
+2. Local Execution Context.
+
+Below is a table that illustrates your Global Execution Context before code execution.
+
+![Global Execution Context before you execute js code](global_execution_context_before_executing_js_code.jpg)
+
+For each line, there will be a visual representation of what the Global Execution Context looks like at that moment.
+
+Line 1: Save in memory a constant, **num**, as the label for an integer, **3**.
+
+![Global Execution Context when line one is executed](step_one_line_one.jpg)
+
+Line 2: You meet a JavaScript keyword [**function**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions). The function keyword means, save in memory **multiplyBy2**, as the label for the functionality of the function, `multiplyBy2()` (denoted as -f- in the image).
+
+This process is also known as **defining/declaring** a function.
+
+![Global Execution Context when line two is executed](step_one_line_two.jpg)
+
+At this point, if you were to run:
+```javascript
+console.log(multiplyBy2.toString());
+```
+You would the source code for `multiplyBy2()`.
+
+To understand what `toString()` does see [the MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/API/Location/toString)
+
+From line 2, you would then move on to line 6 and not line 3. 
+
+Why?
+
+This is because in JavaScript you **do not** go into the body of a function until it is called. And our function `multiplyBy2()` in this case, is defined, but never called.
```

Comment:
As stated on lines 40-44 you may want to combine this section into a paragraph, much like what I suggested on lines 40-44